### DISCUSSION TITLE
Limit on reading nodes?

### DISCUSSION THREAD

**Discussion by TheFern2** [gopcua/opcua#594] [Q&A] [answered]
I've tested opcua async python and my server can handle one request reads 500, 1000, 1500, and 2000.

With go lib I made a ua.ReadValueId array and passed it to the ua.ReadRequest I can read a handful of items, but I tried 500 and instantly got a read timeout request.

What's the best way to read lots of nodes? My server can handle it.

**Comment by magiconair** [comment, marked as answer]
You might need to adjust the `ReadBufferSize`, `WriteBufferSize` and the `MaxChunkCount` to allow larger packages.

**Reply by TheFern2** [reply]
Thanks @magiconair I had forgotten about this question. My plan was to sway my boss to let me do a custom opc client project in go, but they stuck with python and I ended up just using python for the entire project. I no longer work there, but the answer is still useful for future projects.
